lib/public/views/Runs/RunPerPeriod/RunsPerLhcPeriodOverviewModel.js

```diff
@@ -63,11 +63,93 @@ export class RunsPerLhcPeriodOverviewModel extends RunsOverviewModel {
         return this._lhcPeriodName;
     }
 
+    /**
+     * Set name of current lhc period which runs are fetched
+     *
+     * @param {string} lhcPeriodName name of LHC period
+     */
+    set lhcPeriodName(lhcPeriodName) {
+        this._lhcPeriodName = lhcPeriodName;
+        this._tabbedPanelModel.lhcPeriodName = lhcPeriodName;
+    }
+
     /**
      * Get all detectors
+     *
      * @return {RemoteData<Detector[]>} detectors
      */
     get detectors() {
         return this._detectors$.getCurrent();
     }
+
+    /**
+     * Returns the model for the tabbed component
+     *
+     * @return {RunsPerLhcPeriodTabbedPanelModel} the tabbed component model
+     */
+    get tabbedPanelModel() {
+        return this._tabbedPanelModel;
+    }
+}
+
+/**
+ * RunsPerLhcPeriodTabbedPanelModel
+ */
+class RunsPerLhcPeriodTabbedPanelModel extends TabbedPanelModel {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super(Object.values(RUNS_PER_LHC_PERIOD_PANELS_KEYS));
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritDoc
+     */
+    _fetchCurrentPanelData() {
+        switch (this.currentPanelKey) {
+            case RUNS_PER_LHC_PERIOD_PANELS_KEYS.DETECTOR_QUALITIES:
+                this.currentPanelData = null;
+                break;
+            case RUNS_PER_LHC_PERIOD_PANELS_KEYS.SYNCHRONOUS_FLAGS:
+                this._fetchSynchronousQcSummary();
+                break;
+        }
+    }
+
+    /**
+     * Fetch QC summary for synchronous QC flags
+     *
+     * @return {Promise<void>} resolved once data are fetched
+     */
+    async _fetchSynchronousQcSummary() {
+        if (this._lhcPeriodName) {
+            this.currentPanelData = RemoteData.loading();
+            this.notify();
+            try {
+                const { data: [lhcPeriod] } = await jsonFetch(`/api/lhcPeriodsStatistics?filter[names][]=${this._lhcPeriodName}`);
+                if (!lhcPeriod) {
+                    this.currentPanelData = RemoteData.failure([{ title: `Cannot find LHC period with name '${this._lhcPeriodName}'` }]);
+                    this.notify();
+                    return;
```

Comment:
I think that you should use else here, to benefit from the global notify? I think the code would be a bit more uniform. As you prefer.

Reply:
right, applied